# r = open('file.txt', 'a')# отрываем новый фаил(содаем, если нет)
#
# try:
#     r.write('something' + '\n')#записали в фаил какието данные
#     10/0#ошибка
#     r.write("что-то")#еще раз хотим что то записать( но не сможем тк произошла ошибка выше)
# finally:
#     r.close()#безопастное закрытие файла
# print('Все хорошо!')

#с with-as
#если в процессе работы программы возникнет ошибка, программа закроется в любом случае безопастно
#и записанные ранее данные не будут потеряны)

with open('file.txt', 'a') as r:# c функцией open октрыть ее как r (то есть мы присваиваем работу переменной в фунцию r)то же самое что и первая строка
    r.write('something' + '\n')  # записали в фаил какието данные
    10/0#ошибка произойдет но фаил закроется безопастно
    r.write("что-то")#еще раз хотим что то записать( но не сможем тк произошла ошибка выше)
    #здесь мы не пишем r.close, т.к здесь он закрывается автоматически
print('Все норм')

